### PR TITLE
Clip filter

### PR DESCRIPTION
**Pull request by bnmajor**
Extend the existing clipping capabilities to include more modules.

### CODE REVIEW

File: tomviz/modules/Module.cxx [hunk 1]
```diff
@@ -335,4 +335,7 @@ bool Module::setVisibility(bool val) {
   return true;
 }
 
+bool Module::updateClippingPlane(vtkPlane* plane, bool newFilter) {
```
Comment (cryos) [inline tomviz/modules/Module.cxx:338]: Always put function open braces on a new line

File: tomviz/modules/ModuleClip.cxx [hunk 2]
```diff
@@ -90,26 +92,41 @@ bool ModuleClip::initialize(DataSource* data, vtkSMViewProxy* vtkView)
...
   return widgetSetup;
 }
 
-// Should only be called from initialize after ClipVolume has been setup
+// Should only be called from initialize after Clip has been setup
 bool ModuleClip::setupWidget(vtkSMViewProxy* vtkView)
 {
-  vtkAlgorithm* clipVolumeAlg =
-    vtkAlgorithm::SafeDownCast(m_clipVolume->GetClientSideObject());
+  vtkAlgorithm* clipAlg =
```
Comment (cryos) [inline tomviz/modules/ModuleClip.cxx:122]: Use auto?

File: tomviz/modules/ModuleContour.cxx [hunk 3]
```diff
@@ -506,4 +506,16 @@ void ModuleContour::onColorByArrayNameChanged(const QString& name)
   emit renderNeeded();
 }
 
+bool ModuleContour::updateClippingPlane(vtkPlane* plane, bool newFilter) {
```
Comment (cryos) [inline tomviz/modules/ModuleContour.cxx:509]: New line for opening brace of function implementation

File: tomviz/modules/ModuleClip.cxx [hunk 4]
```diff
@@ -587,6 +652,52 @@ void ModuleClip::onPlaneChanged(double* point)
...
+  m_widget->GetNormal(normal);
+  for (auto i = 0; i < 3; ++i) {
+    normal[i] *= -1;
+  }
+  m_widget->SetNormal(normal);
+  m_clippingPlane->SetNormal(normal);
+  onPropertyChanged();
+}
+
+void ModuleClip::onUpdateColor(const QColor& color) {
```
Comment (cryos) [inline tomviz/modules/ModuleClip.cxx:666]: These should probably all just be caught by clang-format, but opening braces get their own line for function level.

File: tomviz/modules/ModuleClip.cxx (hunk 2 above)
Comment (bnmajor) [reply tomviz/modules/ModuleClip.cxx]: Fixed

**Comment by bnmajor** [comment]
@cryos Ran clang-format, the formatting should be corrected now

**Pull request merged**